aws-cpp-sdk-core/source/endpoint/internal/AWSEndpointAttribute.cpp

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/core/endpoint/internal/AWSEndpointAttribute.h>
+#include "aws/core/utils/logging/LogMacros.h"
+
+static const char ENDPOINT_AUTH_SCHEME_TAG[] = "EndpointAuthScheme::BuildEndpointAuthSchemeFromJson";
+
+Aws::Internal::Endpoint::EndpointAttributes
+Aws::Internal::Endpoint::EndpointAttributes::BuildEndpointAttributesFromJson(const Aws::String& iJsonStr)
+{
+    Aws::Internal::Endpoint::EndpointAttributes attributes;
+    Aws::Internal::Endpoint::EndpointAuthScheme& authScheme = attributes.authScheme;
+
+    Utils::Json::JsonValue jsonObject(iJsonStr);
+    if (jsonObject.WasParseSuccessful())
+    {
+        Aws::Map<Aws::String, Utils::Json::JsonView> jsonMap = jsonObject.View().GetAllObjects();
+        for (const auto& mapItemAttribute : jsonMap)
+        {
+            if (mapItemAttribute.first == "authSchemes" && mapItemAttribute.second.IsListType()) {
+                Aws::Utils::Array<Utils::Json::JsonView> jsonAuthSchemeArray = mapItemAttribute.second.AsArray();
+
+                for (size_t arrayIdx = 0; arrayIdx < jsonAuthSchemeArray.GetLength(); ++arrayIdx)
+                {
+                    const Utils::Json::JsonView& property = jsonAuthSchemeArray.GetItem(arrayIdx);
```

Comment:
any reason not to do

```
for(const auto &property: jsonAuthSchemeArray) {
  ...
}
```

Reply:
Aws::Utils::Array does not provide std:: range-for required functions.